scripts/random_scalarizations.py

```diff
@@ -766,7 +767,13 @@ def main(config, black_box_function=None, output_file=""):
     optimization_function_parameters['model_type'] = model_type
     iteration_number = 0
     bo_t0 = datetime.datetime.now()
-    while iteration_number < optimization_iterations:
+    # Measure elapsed time in hours after each iteration
+    run_time = (datetime.datetime.now() - start_time).total_seconds() / 3600
+    # run_time / time_budget < 1 if budget > elapsed time or budget == -1
+    if time_budget > 0:
+        print('starting optimization phase, limited to run for ', time_budget, ' hours')
+
+    while iteration_number < optimization_iterations and run_time / time_budget < 1:
```

Comment:
Yes it will crash before the optimization begin. But that is if the user manually sets time_budget to 0 which would be strange. Can be resolved by something like:

elif time_budget == 0:
    print('Time budget cannot be zero. To not limit runtime set time_budget = -1')
    sys.exit()

Reply:
Yes, I think it would be nice to add the check. I like your code suggestion.

Reply:
commited this check